src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -66,6 +66,7 @@ class Container implements ResettableContainerInterface
     protected $services = array();
     protected $methodMap = array();
     protected $privates = array();
+    protected $synthetics = array();
```

Comment:
we don't need this is we don't put synthetic services in `$this->methodMap`, should allow simplifying many things